src/components/KTreeList/KTreeList.vue

```diff
@@ -0,0 +1,170 @@
+ * lines switch to `v-slot:` notation and Save to let linter clear -->
+ */
+import { computed, ref, watch, onMounted, PropType } from 'vue'
+import KTreeDraggable from '@/components/KTreeList/KTreeDraggable.vue'
+import { TreeListItem } from '@/components/KTreeList/KTreeItem.vue'
+
+export interface ChangeEvent {
+  items: TreeListItem[]
+}
+
+export interface ChildChangeEvent {
+  parent: string,
+  children: TreeListItem[]
+}
+
+const props = defineProps({
+  modelValue: {
+    type: Array as PropType<TreeListItem[]>,
+    default: null,
+    validator: (items: TreeListItem[]) => !items.length || items.every(i => i.name !== undefined && i.id !== undefined),
+  },
+  items: {
+    type: Array as PropType<TreeListItem[]>,
+    default: null,
+    validator: (items: TreeListItem[]) => !items.length || items.every(i => i.name !== undefined && i.id !== undefined),
+  },
+  disableDrag: {
+    type: Boolean,
+    default: false,
+  },
+  maxLevel: {
+    type: Number,
+    default: 2,
+  },
+})
+
+const emit = defineEmits<{
+  (event: 'change', data: ChangeEvent): void,
+  (event: 'child-change', data: ChildChangeEvent): void,
+  (event: 'selected', item: TreeListItem): void
+}>()
+
+// const key = ref(0)
+const internalList = ref<TreeListItem[]>([])
+
+// we need this so we can create a watcher for programmatic changes to the modelValue
+const value = computed({
+  get(): TreeListItem[] {
+    return props.modelValue
+  },
+  set(newValue: TreeListItem[]): void {
+    internalList.value = newValue
+  },
+})
+
+const handleSelection = (itemToSelect: TreeListItem, list?: TreeListItem[]): void => {
+  const editList = list || internalList.value
+
+  editList.forEach((item: TreeListItem) => {
+    if (item.id === itemToSelect.id) {
+      item.selected = true
+    } else {
+      item.selected = false
+    }
+
+    if (item.children?.length) {
+      handleSelection(itemToSelect, item.children)
+    }
+  })
+  emit('selected', itemToSelect)
+}
+
+const handleChangeEvent = (data: ChangeEvent): void => {
+  emit('change', data)
+}
+
+const handleChildChangeEvent = (data: ChildChangeEvent): void => {
+  emit('child-change', data)
+}
+
+// watch for programmatic changes to modelValue
+watch(value, (newVal, oldVal) => {
+  if (JSON.stringify(newVal) !== JSON.stringify(oldVal)) {
+    internalList.value = newVal
+    internalList.value.forEach((item: TreeListItem) => {
+      if (!item.children) {
+        item.children = []
+      }
+    })
+  }
+})
+
+watch(() => props.items, (newValue, oldValue) => {
+  // Only trigger the watcher if items actually change
+  if (JSON.stringify(newValue) !== JSON.stringify(oldValue)) {
+    internalList.value = newValue
+    internalList.value.forEach((item: TreeListItem) => {
+      if (!item.children) {
+        item.children = []
+      }
+    })
+  }
+})
+
+onMounted(() => {
+  if (props.modelValue) {
+    internalList.value = props.modelValue
+  } else if (props.items) {
+    internalList.value = props.items
+  }
```

Comment:
Should we `console.warn` if the user defines both `v-model` (or `modelValue`) along with `items`? 

Everything would still work as you have it, but we could say something like:

```ts
console.warn('KTreeList: You should not provide both v-model (or props.modelValue) along with props.item')
```